### DISCUSSION TITLE
CI/CD doesn't publish artifacts in custom branches

### DISCUSSION THREAD

**Discussion by cegekaJG** [microsoft/AL-Go#1582] [Ideas]
One of our repos is now running CI/CD for a secondary branch, but I noticed that the CI/CD workflows from this branch aren't creating any artifacts of the compiled apps, even though they encounter no errors and the default branch publishes its apps just fine.

Here's the Build log from the default (`main`) branch:
[logs_36.zip](https://github.com/microsoft/AL-Go/files/14379092/logs_36.zip)

And here's the Build log from the second (`testmaster`) branch:
[logs_37.zip](https://github.com/microsoft/AL-Go/files/14379093/logs_37.zip)

Both branches have the same AL Go system files. This behavior is consistent with a different repository using the same settings and branching structure. I know I have `"doNotPublishApps"` enabled, but that never stopped the apps on `main` being published.

**Comment by freddydk** [comment]
Currently, artifacts are being created for the main branch, for release branches and when it is detected that there are deliveryTargets or DeploymentTargets, controlled by this line in CI/CD:
```
publishArtifacts: ${{ github.ref_name == 'main' || startswith(github.ref_name, 'release/') || startswith(github.ref_name, 'releases/') || needs.Initialization.outputs.deliveryTargetsJson != '[]' || needs.Initialization.outputs.environmentCount > 0 }}
```
We did earlier discuss always publishing artifacts and maybe that is a better choice.

**Comment by cegekaJG** [comment]
At the very least, being able to define custom glob or regex patterns for this would be extremely useful - we have a couple client-agnostic apps in their own repository, and since they are only accessed using `"appDependencyProbingPaths"`, there's no need to define environments of delivery targets.
